app/models/invitation.rb

```diff
@@ -0,0 +1,14 @@
+class Invitation < ActiveRecord::Base
+  def self.find_by_slug(slug)
+    Invitation
+      .where(slug: slug)
+      .where(Invitation.arel_table[:expiration_date].gt(Date.today))
+      .take!
+  rescue RangeError
+    raise RecordNotFound.new('The invitation does not exist or it has expired')
```

Comment:
If you add the invitation-organization relation - see my next comment - you can rewrite this method quite easy: 

```ruby
class Organization
      def active_invitation
           invitations.first('expiration_date < ?', Date.today)
      end
end
```

Then just do `Organization.current.active_invitation` to get the current invitation, and then something like `invitation.active?` that checks if the given invitation is the active of this organization.